client/src/components/common/boost/BoostConfigurator.tsx

```diff
@@ -26,84 +33,134 @@ export default function BoostConfigurator({
   onUploadConfig,
 }: BoostConfiguratorProps) {
   const fileInput = createRef<HTMLInputElement>();
-  const [configType, setConfigType] = useState<BoostConfigType>('all');
+  const [configType, setConfigType] = useState<FileConfigT>('bundle');
+
+  const [displayMessage, setDisplayMessage] = useState('');
+
+  const [showUploadErr, setShowUploadErr] = useState(false);
+  const handleErrorClose = () => setShowUploadErr(false);
 
   // Function to click the hidden file input button (this is a work around to avoid the ugly
   // UI of the default input file button)
-  const handleClickUpload = (type: BoostConfigType) => {
+  const handleClickUpload = (type: FileConfigT) => {
     if (fileInput.current) {
       setConfigType(type);
       fileInput.current.click();
     }
   };
 
+  // Check if a file with the same given name exists
+  const fileExist = (name: string) => {
+    let found = false;
+    configs.forEach((config) => {
+      config.options.forEach((configName) => {
+        if (configName.fileName === name) {
+          found = true;
+        }
+      });
+    });
+    return found;
+  };
```

Comment:
Could use destructuring, [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) and [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) here to make this more understandable.

Reply:
It's a bit tough using those methods because there's no simply array here and if I try using them I doubt it's going to be just as confusing. For example, you can't really use `includes` because we're comparing an object's `fileName` attribute with a string, and so an array of the object will not 'include' a string. Maybe there's a way around this in javaScript, but I can't seem to find it.

Reply:
Ah, yes you're right, `includes` is not suitable here because `options` is an array of objects. My point still stands without `includes`.

```typescript
const fileExist2 = (name: string) =>
    Boolean(
      configs.find(({ options }) =>
        options.find(({ fileName }) => fileName === name),
      ),
    );
```

Reply:
Even then, the original code could utilise destructuring to be more readable

```typescript
const fileExist = (name: string) => {
    let found = false;
    configs.forEach(({ options }) => {
      options.forEach(({ fileName }) => {
        if (fileName === name) found = true;
      });
    });
    return found;
  };
```

Reply:
Tbh I don't see a big difference in readability. My now coding style is quite basic (like I'd rather do a full if statement:
```
if (fileName ===name) {
found = true;
}
```
rather than
```
if (fileName ===name) found = true;
```
So the `Boolean()` magic actually makes readability harder for me. Happy to change it to whichever version you think is more readable for you though. Just let me know!

Reply:
That's really not the point. The point was that we could use destructuring to avoid long chains. Feel free to use whatever if statement format you feel comfortable with.

On `Boolean()`: It's converting it to a boolean, which is exactly what the function is supposed to do.